### PR TITLE
Fix data race in gRPC client

### LINKED ISSUE #26573: [Bug]: Data race in gRPC client
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

image

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
/kind bug
fix #26573

### CODE REVIEW

File: internal/util/grpcclient/client.go [hunk 1]
```diff
@@ -110,6 +110,8 @@ func (c *ClientBase[T]) GetRole() string {
 
 // GetAddr returns address of client
 func (c *ClientBase[T]) GetAddr() string {
+	c.grpcClientMtx.RLock()
+	defer c.grpcClientMtx.RUnlock()
 	return c.addr
```
Comment (xiaofan-luan) [inline internal/util/grpcclient/client.go:115]: probably change addr to a atomic?

Comment (xiaofan-luan) [reply internal/util/grpcclient/client.go:115]: becasue there are actually many different places we use addr

**Comment by yah01** [comment]
Also add a constructor for `ClientBase`, simplify client code

**Comment by congqixia** [comment]
manual pass for blocking other prs
all code checkers passed except amazon linux image due to #26626

**Pull request merged**